src/Deloser.ts

```diff
@@ -340,17 +341,27 @@ function buildSelector(element: HTMLElement): string | undefined {
 
     const selector: string[] = [buildElementSelector(element)];
 
-    let el = element.parentElement;
+    let node = dom.getParentNode(element);
 
-    while (el) {
-        const isBody = el.tagName === "BODY";
-        selector.unshift(buildElementSelector(el, false, !isBody));
-
-        if (isBody) {
+    while (node) {
```

Comment:
```suggestion
    while (node &&  node.nodeType !== Node.DOCUMENT_FRAGMENT_NODE) {
```

this should remove the need to check the nodeType inside the loop?